bluesky/plan_patterns.py

```diff
@@ -67,6 +67,99 @@ def spiral(x_motor, y_motor, x_start, y_start, x_range, y_range, dr, nth, *,
     return cyc
 
 
+def spiral_square_pattern(x_motor, y_motor, x_center, y_center, x_range, y_range, x_num,y_num):
+    '''Square spiral scan, centered around (x_start, y_start)
+
+    Parameters
+    ----------
+    x_motor : object, optional
+        any 'setable' object (motor, temp controller, etc.)
+    y_motor : object, optional
+        any 'setable' object (motor, temp controller, etc.)
+    x_center : float
+        x center
+    y_center : float
+        y center
+    x_range : float
+        x width of spiral
+    y_range : float
+        y width of spiral
+    x_num : float
+        number of x axis points
+    y_num : float (must be even if x_num is even and must be odd if x_num is odd, if not it is increased by 1 to ensure this)
```

Comment:
This I found to be important because of the path required for the square spiral, if they are both even then there is no 'center' square but the first 4 positions 'surround' the square. If they are both odd then the first point is the centre square

e.g. for both even the centre looks like this (with px being the x'th position):

p7   p8  p9   p10
p6   p1   p2   p11
p5   p4  p3   p12
p16 p15 p14 p13

If they are both odd the centre looks like this:

p7  p8   p9
p6  p1   p2
p5  p4  p3

I don't think there is any path (at least I could not come up with any path) to 'make' a square spiral work if one is odd and the other is even.

Hopefully this explains the issue, if not we should discuss in person where we can make some drawings/diagrams .